memcached 3.1.3 does not compile on PHP 8.0

Hello,
we are running memcached in our CI system and have a daily PHP image for the newer versions and unfortunately the memcached extension doesn't compile anymore for PHP 8.0.

Our docker image file is here: https://github.com/joomla-projects/docker-images/tree/php8.0

And the error is 
```
/usr/src/php/ext/memcached/php_memcached_session.c: In function 'ps_read_memcached':
/usr/src/php/ext/memcached/php_memcached_session.c:448:25: error: expected ')' before 'TSRMLS_CC'
   php_error_docref(NULL TSRMLS_CC, E_WARNING, "error getting session from memcached: %s", memcached_last_error_message(memc));
                         ^~~~~~~~~
/usr/src/php/ext/memcached/php_memcached_session.c:448:3: error: too few arguments to function 'php_error_docref'
   php_error_docref(NULL TSRMLS_CC, E_WARNING, "error getting session from memcached: %s", memcached_last_error_message(memc));
   ^~~~~~~~~~~~~~~~
In file included from /usr/src/php/ext/memcached/php_memcached.h:20,
                 from /usr/src/php/ext/memcached/php_memcached_session.c:17:
/usr/local/include/php/main/php.h:342:23: note: declared here
 PHPAPI ZEND_COLD void php_error_docref(const char *docref, int type, const char *format, ...)
                       ^~~~~~~~~~~~~~~~
/usr/src/php/ext/memcached/php_memcached_session.c: In function 'ps_write_memcached':
/usr/src/php/ext/memcached/php_memcached_session.c:478:26: error: expected ')' before 'TSRMLS_CC'
    php_error_docref(NULL TSRMLS_CC, E_WARNING, "error saving session to memcached: %s", memcached_last_error_message(memc));
                          ^~~~~~~~~
/usr/src/php/ext/memcached/php_memcached_session.c:478:4: error: too few arguments to function 'php_error_docref'
    php_error_docref(NULL TSRMLS_CC, E_WARNING, "error saving session to memcached: %s", memcached_last_error_message(memc));
    ^~~~~~~~~~~~~~~~
In file included from /usr/src/php/ext/memcached/php_memcached.h:20,
                 from /usr/src/php/ext/memcached/php_memcached_session.c:17:
/usr/local/include/php/main/php.h:342:23: note: declared here
 PHPAPI ZEND_COLD void php_error_docref(const char *docref, int type, const char *format, ...)
                       ^~~~~~~~~~~~~~~~
make: *** [Makefile:206: php_memcached_session.lo] Error 1
```

--- a/.travis/travis.sh
+++ b/.travis/travis.sh
@@ -67,8 +67,9 @@ function install_libmemcached() {
 }
 
 function install_igbinary() {
-    git clone https://github.com/igbinary/igbinary.git
+    git clone https://github.com/sodabrew/igbinary.git
     pushd igbinary
+        git checkout php-8
         phpize
         ./configure
         make

--- a/php_memcached_session.c
+++ b/php_memcached_session.c
@@ -445,7 +445,7 @@ PS_READ_FUNC(memcached)
 		*val = ZSTR_EMPTY_ALLOC();
 		return SUCCESS;
 	} else {
-		php_error_docref(NULL TSRMLS_CC, E_WARNING, "error getting session from memcached: %s", memcached_last_error_message(memc));
+		php_error_docref(NULL, E_WARNING, "error getting session from memcached: %s", memcached_last_error_message(memc));
 		return FAILURE;
 	}
 }
@@ -475,7 +475,7 @@ PS_WRITE_FUNC(memcached)
 		if (memcached_set(memc, key->val, key->len, val->val, val->len, expiration, 0) == MEMCACHED_SUCCESS) {
 			return SUCCESS;
 		} else {
-			php_error_docref(NULL TSRMLS_CC, E_WARNING, "error saving session to memcached: %s", memcached_last_error_message(memc));
+			php_error_docref(NULL, E_WARNING, "error saving session to memcached: %s", memcached_last_error_message(memc));
 		}
 	} while (--retries > 0);
 

